packages/e2e-tests/plugins/meta-attribute-block/index.js

```diff
@@ -7,25 +7,21 @@
 		icon: 'star',
 		category: 'common',
 
-		attributes: {
-			content: {
-				type: "string",
-				source: "meta",
-				meta: "my_meta",
-			},
-		},
-
 		edit: function( props ) {
-			return el(
-					'input',
-					{
-						className: 'my-meta-input',
-						value: props.attributes.content,
-						onChange: function( event ) {
-							props.setAttributes( { content: event.target.value } );
-						},
-					}
-			);
+			var dispatch = wp.data.useDispatch();
+			return el( 'input', {
+				className: 'my-meta-input',
+				value: wp.data.useSelect(
+					( select ) =>
+						select( 'core/editor' ).getEditedEntityAttribute( 'meta' ).my_meta,
```

Comment:
A block shouldn't depend on the editor store or the editor module. How can we use the entity provider to edit that meta?

Reply:
Good point, here: https://github.com/WordPress/gutenberg/pull/17020/commits/0bfcf28198679619aa3eab10ca1a7d2f2e3916ed

I refactored it so that the block editor calls simple hooks, but the editor provides the implementation, if any.